ethereumj-core/src/main/java/org/ethereum/sync/SyncManager.java

```diff
@@ -63,20 +63,14 @@
     // Transaction.getSender() is quite heavy operation so we are prefetching this value on several threads
     // to unload the main block importing cycle
     private ExecutorPipeline<BlockWrapper,BlockWrapper> exec1 = new ExecutorPipeline<>
-            (4, 1000, true, new Functional.Function<BlockWrapper,BlockWrapper>() {
-                public BlockWrapper apply(BlockWrapper blockWrapper) {
-                    for (Transaction tx : blockWrapper.getBlock().getTransactionsList()) {
-                        tx.getSender();
-                    }
-                    return blockWrapper;
-                }
-            }, new Functional.Consumer<Throwable>() {
-                public void accept(Throwable throwable) {
-                    logger.error("Unexpected exception: ", throwable);
+            (4, 1000, true, blockWrapper -> {
+                for (Transaction tx : blockWrapper.getBlock().getTransactionsList()) {
+                    tx.getSender();
```

Comment:
Let's party hard :)
```java
(4, 1000, true, blockWrapper -> blockWrapper.getBlock().getTransactionsList().forEach(Transaction::getSender));
```

Reply:
Ahh, yeah, not the only thing that could be upgraded to streams, didn't try it. Big enough commit already )